### PR TITLE
Validation support for properties kafka file

### LINKED ISSUE #152: Validation support for properties kafka file

### PR DESCRIPTION
**Pull request by angelozerr**
Validation support for properties kafka file

Fixes #152

Signed-off-by: azerr <azerr@redhat.com>

### CODE REVIEW

**Comment by angelozerr** [comment]
Here a demo:

DiagnosticsDemo2

**Comment by fbricon** [comment]
that looks really cool!

Small nitpick: the diagnostic range shouldn't include the leading space of the bad value

**Comment by angelozerr** [comment]
> that looks really cool!

Glad it pleases you:)

> Small nitpick: the diagnostic range shouldn't include the leading space of the bad value

You are right, I must improve that.

**Comment by angelozerr** [comment]
> Small nitpick: the diagnostic range shouldn't include the leading space of the bad value

Fixed!

**Review by fbricon — CHANGES_REQUESTED** [review]

File: src/test/suite/kafka-file/languageservice/diagnosticsProperties.test.ts [hunk 1]
```diff
@@ -0,0 +1,365 @@
+import { DiagnosticSeverity } from "vscode";
+import { assertDiagnostics, diagnostic, position } from "./kafkaAssert";
+
+suite("Kafka File Diagnostics Test Suite", () => {
+
+    test("Empty diagnostics", async () => {
+        await assertDiagnostics('', []);
+    });
+});
+
```
[the remark this answers is not in the document]
Comment (angelozerr) [reply src/test/suite/kafka-file/languageservice/diagnosticsProperties.test.ts:10]: Yes sure I will implement it.

File: docs/Consuming.md [hunk 2]
```diff
@@ -89,6 +89,14 @@ Completion is available for
 
 ![Topic completion](assets/kafka-file-consumer-topic-completion.png)
 
+#### Validation
+
+Validation will help you to write valid kafka file.
```
[the remark this answers is not in the document]
Comment (angelozerr) [reply docs/Consuming.md]: fixed

File: docs/Consuming.md [hunk 3]
```diff
@@ -89,6 +89,14 @@ Completion is available for
 
 ![Topic completion](assets/kafka-file-consumer-topic-completion.png)
 
+#### Validation
+
+Validation will help you to write valid kafka file.
+
+Here a sample with topic syntax:
```
[the remark this answers is not in the document]
Comment (angelozerr) [reply docs/Consuming.md]: fixed

File: docs/Producing.md [hunk 4]
```diff
@@ -67,6 +67,14 @@ Completion is available for
  
 ![Topic completion](assets/kafka-file-producer-topic-completion.png)
 
+### Validation
+
+Validation will help you to write valid kafka file.
```
[the remark this answers is not in the document]
Comment (angelozerr) [reply docs/Producing.md]: fixed

File: docs/Producing.md [hunk 5]
```diff
@@ -67,6 +67,14 @@ Completion is available for
  
 ![Topic completion](assets/kafka-file-producer-topic-completion.png)
 
+### Validation
+
+Validation will help you to write valid kafka file.
+
+Here a sample with producer which doesn't define the value:
```
[the remark this answers is not in the document]
Comment (angelozerr) [reply docs/Producing.md]: fixed

File: src/test/suite/kafka-file/languageservice/diagnosticsProperties.test.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (angelozerr) [reply src/test/suite/kafka-file/languageservice/diagnosticsProperties.test.ts:10]: > can you test duplicate properties?

implemented (with test)

**Comment by fbricon** [comment]
Thanks @angelozerr!

**Pull request merged**